webapp/components/youtube_video.jsx

```diff
@@ -120,53 +119,65 @@ export default class YoutubeVideo extends React.Component {
-                        />
-                        <div className='block'>
-                            <span className='play-button'><span/></span>
+        if (this.state.loaded) {
+            let header = 'Youtube';
+            if (this.state.title) {
+                header = header + ' - ';
+            }
+
+            let content;
+            if (this.state.playing) {
+                content = (
+                    <iframe
+                        src={'https://www.youtube.com/embed/' + this.state.videoId + '?autoplay=1&autohide=1&border=0&wmode=opaque&fs=1&enablejsapi=1' + this.state.time}
+                        width='480px'
+                        height='360px'
+                        type='text/html'
+                        frameBorder='0'
+                        allowFullScreen='allowfullscreen'
+                    />
+                );
+            } else {
+                content = (
+                    <div className='embed-responsive embed-responsive-4by3 video-div__placeholder'>
+                        <div className='video-thumbnail__container'>
+                            <img
+                                className='video-thumbnail'
+                                src={'https://i.ytimg.com/vi/' + this.state.videoId + '/hqdefault.jpg'}
+                            />
+                            <div className='block'>
+                                <span className='play-button'><span/></span>
+                            </div>
                         </div>
                     </div>
+                );
+            }
+
+            return (
+                <div>
+                    <h4>
+                        <span className='video-type'>{header}</span>
+                        <span className='video-title'>
+                            <a
+                                href={this.props.link}
+                                target='blank'
+                                rel='noopener noreferrer'
+                            >
+                                {this.state.title}
+                            </a>
+                        </span>
+                    </h4>
+                    <div
+                        className='video-div embed-responsive-item'
+                        onClick={this.play}
+                    >
+                        {content}
+                    </div>
                 </div>
             );
         }
 
-        return (
-            <div>
-                <h4>
-                    <span className='video-type'>{header}</span>
-                    <span className='video-title'><Link to={this.props.link}>{this.state.title}</Link></span>
-                </h4>
-                <div
-                    className='video-div embed-responsive-item'
-                    onClick={this.play}
-                >
-                    {content}
-                </div>
-            </div>
-        );
+        return <div/>;
```

Comment:
Minor, but can you do
```
if (!this.state.loaded) {
    return <div/>;
}

// return youtube video
```

instead? I think the final return should always be the success case and it also causes a single indentation over a whole bunch :)